### PR TITLE
Replace UpdateOwnership methods with AddOwnership methods

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Here we are replacing UpdateOwnership methods for RBAC resources with AddOwnership and RemoveOwnership methods.
Due the case mentioned in https://github.com/portworx/torpedo/pull/1690#discussion_r1338036296

**Which issue(s) this PR fixes** (optional)
Closes #PA-1678

**Special notes for your reviewer**:
Ive added a sample-test to test these methods, will be removing the test while merging.
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2936/

Existing Automation Runs.
Vannilla-S3:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2938/

RKE-S3
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/s3/job/px-backup-system-test-rke/318/

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -4710,3 +4524,509 @@ func DeletePodWhileRestoreInProgress(ctx context.Context, orgId string, restoreN
...
+	}
+	currentGroupsConfigs := backupLocationInspectResp.BackupLocation.GetOwnership().GetGroups()
+	for _, currentGroupConfig := range currentGroupsConfigs {
+		groupBackupLocationOwnershipAccessConfigs = append(groupBackupLocationOwnershipAccessConfigs, currentGroupConfig)
+	}
+
+	currentUsersConfigs := backupLocationInspectResp.BackupLocation.GetOwnership().GetCollaborators()
+	for _, currentUserConfig := range currentUsersConfigs {
+		userBackupLocationOwnershipAccessConfigs = append(userBackupLocationOwnershipAccessConfigs, currentUserConfig)
+	}
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:4586]: Can't we just append the slice `currentUsersConfigs` to `userBackupLocationOwnershipAccessConfigs` ?

Comment (ak-px) [reply tests/backup/backup_helper.go]: @mkoppal-px , made the changes to append it directly. Thanks for the suggestion

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by snigdha-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Pull request merged**